lib/improver/utilities/temporal_interpolation.py

```diff
@@ -0,0 +1,439 @@
+            ValueError: If interpolation method not in known list.
+
+        """
+        if interval_in_minutes is None and times is None:
+            raise ValueError("TemporalInterpolation: One of "
+                             "'interval_in_minutes' or 'times' must be set. "
+                             "Currently both are none.")
+
+        self.interval_in_minutes = interval_in_minutes
+        self.times = times
+        known_interpolation_methods = ['linear', 'solar', 'daynight']
+        if interpolation_method not in known_interpolation_methods:
+            raise ValueError("TemporalInterpolation: Unknown interpolation "
+                             "method {}. ".format(interpolation_method))
+        self.interpolation_method = interpolation_method
+
+    def __repr__(self):
+        """Represent the configured plugin instance as a string."""
+        result = ('<TemporalInterpolation: interval_in_minutes: {}, '
+                  'times: {}, method: {}>')
+        return result.format(self.interval_in_minutes, self.times,
+                             self.interpolation_method)
+
+    def construct_time_list(self, initial_time, final_time):
+        """
+        A function to construct a list of datetime objects formatted
+        appropriately for use by iris' interpolation method.
+
+        Args:
+            initial_time (datetime.datetime):
+                The start of the period over which a time list is to be
+                constructed.
+            final_time (datetime.datetime).
+                The end of the period over which a time list is to be
+                constructed.
+
+        Returns:
+            list:
+                A list containing a tuple that specifies the coordinate and a
+                list of points along that coordinate to which to interpolate,
+                as required by the iris interpolation method:
+                    e.g. [('time', [<datetime object 0>,
+                                    <datetime object 1>])]
+
+        Raises:
+            ValueError: If list of times provided falls outside the range
+                        specified by the initial and final times.
+            ValueError: If the interval_in_minutes does not divide the time
+                        range up equally.
+        """
+        time_list = []
+        if self.times is not None:
+            self.times = sorted(self.times)
+            if self.times[0] < initial_time or self.times[-1] > final_time:
+                raise ValueError(
+                    'List of times falls outside the range given by '
+                    'initial_time and final_time. ')
+            time_list = self.times
+        else:
+            if ((final_time - initial_time).seconds %
+                    (60 * self.interval_in_minutes) != 0):
+                raise ValueError(
+                    'interval_in_minutes provided to time_interpolate does not'
```

Comment:
Added

Reply:
At the moment, this error message would be:
`interval_in_minutes provided to time_interpolate does not divide into the interval equally.61`. 

I think it would be more readable if it said: 
`The interval_in_minutes of {} does not divide into the interval of {} equally.'.format(self.interval_in_minutes, (final_time - initial_time).seconds)` i.e.
```python
                raise ValueError(
                    'The interval_in_minutes of {} does not divide into '
                    'the interval of {} equally.'.format(
                        self.interval_in_minutes,
                        (final_time - initial_time).seconds))
```

Reply:
Doesn't make a lot of sense having the argument interval_in_minutes and the argument in seconds so I've converted the interval to mins too but the rest of the change is as you requested.